### ISSUE TITLE
spirv-fuzz: Add tests for fuzzer_util.cpp

### ISSUE THREAD

**Issue by Vasniktel** [KhronosGroup/SPIRV-Tools#3493]
`fuzzer_util.cpp` has many functions that change SPIR-V module. These include `Add*` functions (e.g. `AddVectorType` and others). It would be good to have some tests for them

**Comment by afd** [comment]
@Vasniktel Perhaps this could be split into several "good first issue"s.

**Comment by Vasniktel** [comment]
Agreed. However, there are a lot of functions in `fuzzerutil::`. Thus, it might be better to have a single issue for this and instead make multiple PRs to fix it. We could split PRs as follows:
- tests for all `Add*` functions (e.g. `AddIntegerType`, `AddFloatType`, `AddFunctionType`, `AddVariableIdToEntryPointInterfaces`, `AddGlobalVariable`, `AddLocalVariable`, `AddVectorType`, `AddStructType`)
- tests for all `MaybeGet*` functions (e.g. `MaybeGetScalarConstant`, `MaybeGetCompositeConstant`, `MaybeGetIntegerConstant` etc).
- everything else.

There are some trivial functions there, though, which we could probably ignore (e.g. `FloatToWord`, `RepeatedUInt32PairToMap`, `InstructionHasNoSideEffects` and maybe others).

@afd, what do you think?

**Comment by Vasniktel** [comment]
I've created to separate issues for `Add*` functions and `MaybeGet*` functions: #4178 #4179.

Thus, we can focus this issue on writing tests for the following functions:
- `RemoveParameter`
- `UpdateFunctionType`
- `NewTerminatorPreservesDominationRules` (maybe)
